panda/src/navigation/navMesh.h

```diff
@@ -0,0 +1,26 @@
+#ifndef NAVMESH_H
+#define NAVMESH_H
+
+#include "Recast.h"
+#include "DetourNavMesh.h"
+#include "pandaFramework.h"
+
+class NavMesh
+{
```

Comment:
NavMesh should be reference-counted, so that it can be passed between Python and C++ applications without issues.  You can do this by inheriting from ReferenceCount and making sure you allocate it with `new` and pass it by pointer.  Use `PT()` like with other Panda classes whenever storing it to make sure that the object won't get deleted.

We actually want to inherit it from TypedReferenceCount (which adds typing information on top of ReferenceCount, for the benefit of using the class from Python), and add all the TypeHandle boilerplate that can be seen in other Panda classes (look at `panda/src/skel/typedSkel.h` for an example of this.)

Eventually, we will want to change it to inherit from TypedWritableReferenceCount, which will be necessary to add the ability to serialise NavMesh objects from/to .bam files.

These suggestions only apply to NavMesh (and NavMeshNode, but it already inherits from TypedWritableReferenceCount indirectly through PandaNode), since these are the only classes that will need to be serialised to disk.

Reply:
Done in latest commit